### PR TITLE
Code to cleanup log4j remnants post install-oracle

### PR DESCRIPTION
**Pull request by jcnars**
This code has been tested successfully as follows and have been verified to clear 100% of vulnerable log4j jars/wars against a 19.3, 19.17, and 19.18 database (standalone and RAC):
```
✔ ~/mydrive/bmaas/log4j_remediation/bms-toolkit [log4j_kr_remediation|✚ 17…6]
19:53 $ runlocalssh ./install-oracle.sh --ora-swlib-bucket gs://bmaas-testing-oracle-software --instance-ssh-user ansible1 --instance-ssh-key ~/.ssh/id_rsa_bms_toolkit.ansible1_in_poc --instance-ssh-extra-args ''\''-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentityAgent=no -o IdentitiesOnly=true -o ProxyCommand="ssh -W %h:%p -q jcnarasimhan@control-node.mfild.com"'\'''  --backup-dest /u01/backups --ora-swlib-path /u01/oracle_install --ora-version 19 --ora-swlib-type gcs --ora-asm-disks workshop_asm_disk_config.json --ora-data-mounts workshop_data_mounts_config.json --cluster-type RAC --cluster-config workshop_cluster_config.json --ora-data-diskgroup DATA --ora-reco-diskgroup RECO --ora-db-name orcl --ora-db-container false  --no-patch  -- '--tags patch-vulns'      >> /usr/local/google/home/jcnarasimhan/mydrive/bmaas/workshop/bms-toolkit/logs/log4remediation_1
SSO password for jcnarasimhan:
Please touch your security key.
Touch registered. Getting ticket.
✔ ~/mydrive/bmaas/log4j_remediation/bms-toolkit [log4j_kr_remediation|✚ 17…6]
19:54 $
```

The log for the above run is:
```
Command used:
/usr/local/google/home/jcnarasimhan/mydrive/bmaas/log4j_remediation/bms-toolkit/install-oracle.sh --ora-swlib-bucket gs://bmaas-testing-oracle-software --instance-ssh-user ansible1 --instance-ssh-key /usr/local/google/home/jcnarasimhan/.ssh/id_rsa_bms_toolkit.ansible1_in_poc --instance-ssh-extra-args '-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentityAgent=no -o IdentitiesOnly=true -o ProxyCommand="ssh -W %h:%p -q jcnarasimhan@control-node.mfild.com"' --backup-dest /u01/backups --ora-swlib-path /u01/oracle_install --ora-version 19 --ora-swlib-type gcs --ora-asm-disks workshop_asm_disk_config.json --ora-data-mounts workshop_data_mounts_config.json --cluster-type RAC --cluster-config workshop_cluster_config.json --ora-data-diskgroup DATA --ora-reco-diskgroup RECO --ora-db-name orcl --ora-db-container false --no-patch -- --tags patch-vulns

Inventory file for this execution: ./inventory_files/inventory_orcl_RAC.

Running with parameters from command line or environment variables:

ANSIBLE_DISPLAY_SKIPPED_HOSTS=false
ANSIBLE_LOG_PATH=./logs/log_orcl_20230303_195358_RAC.log ARCHIVE_BACKUP_MIN=30
ARCHIVE_ONLINE_DAYS=7
ARCHIVE_REDUNDANCY=2
BACKUP_DEST=/u01/backups
BACKUP_LEVEL0_DAYS=0
BACKUP_LEVEL1_DAYS=1-6
BACKUP_LOG_LOCATION=/home/oracle/logs
BACKUP_REDUNDANCY=2
BACKUP_SCRIPT_LOCATION=/home/oracle/scripts
BACKUP_START_HOUR=01
BACKUP_START_MIN=00
CLUSTER_CONFIG=workshop_cluster_config.json
CLUSTER_TYPE=RAC
INSTANCE_HOSTGROUP_NAME=dbasm
INSTANCE_HOSTNAME=
INSTANCE_IP_ADDR=
INSTANCE_SSH_EXTRA_ARGS=''\''-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentityAgent=no -o IdentitiesOnly=true -o ProxyCommand="ssh -W %h:%p -q jcnarasimhan@control-node.mfild.com"'\''' INSTANCE_SSH_KEY=/usr/local/google/home/jcnarasimhan/.ssh/id_rsa_bms_toolkit.ansible1_in_poc INSTANCE_SSH_USER=ansible1
ORA_ASM_DISKS=workshop_asm_disk_config.json
ORA_DATA_DISKGROUP=DATA
ORA_DATA_MOUNTS=workshop_data_mounts_config.json
ORA_DB_CHARSET=AL32UTF8
ORA_DB_CONTAINER=false
ORA_DB_DOMAIN=
ORA_DB_NAME=orcl
ORA_DB_NCHARSET=AL16UTF16
ORA_DB_TYPE=MULTIPURPOSE
ORA_DISK_MGMT=UDEV
ORA_EDITION=EE
ORA_LISTENER_NAME=LISTENER
ORA_LISTENER_PORT=1521
ORA_PDB_COUNT=1
ORA_PDB_NAME_PREFIX=PDB
ORA_RECO_DISKGROUP=RECO
ORA_REDO_LOG_SIZE=100MB
ORA_RELEASE=base
ORA_ROLE_SEPARATION=TRUE
ORA_STAGING=/u01/oracle_install
ORA_SWLIB_BUCKET=gs://bmaas-testing-oracle-software ORA_SWLIB_CREDENTIALS=
ORA_SWLIB_PATH=/u01/oracle_install
ORA_SWLIB_TYPE=gcs
ORA_VERSION=19.3.0.0.0
PB_CHECK_INSTANCE=check-instance.yml
PB_CONFIG_DB=config-rac-db.yml
PB_CONFIG_RAC_DB=config-rac-db.yml
PB_INSTALL_SW=install-sw.yml
PB_LIST='check-instance.yml prep-host.yml install-sw.yml config-rac-db.yml' PB_PREP_HOST=prep-host.yml
PRIMARY_IP_ADDR=

Ansible params: --tags patch-vulns
Found Ansible: ansible-playbook is /usr/bin/ansible-playbook

Running Ansible playbook: ansible-playbook -i ./inventory_files/inventory_orcl_RAC  --tags patch-vulns check-instance.yml

PLAY [dbasm] *******************************************************************

PLAY RECAP *********************************************************************

Running Ansible playbook: ansible-playbook -i ./inventory_files/inventory_orcl_RAC  --tags patch-vulns prep-host.yml

PLAY [Populate variables] ******************************************************

PLAY [dbasm] *******************************************************************

TASK [Gathering Facts] ********************************************************* ok: [g264413522-s313]
ok: [g264413522-s314]

PLAY [dbasm[0]] ****************************************************************

TASK [Gathering Facts] ********************************************************* ok: [g264413522-s313]

PLAY RECAP *********************************************************************
g264413522-s313            : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
g264413522-s314            : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Running Ansible playbook: ansible-playbook -i ./inventory_files/inventory_orcl_RAC  --tags patch-vulns install-sw.yml

PLAY [Populate variables] ******************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm[0]] ****************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm[0]] ****************************************************************

PLAY RECAP *********************************************************************

Running Ansible playbook: ansible-playbook -i ./inventory_files/inventory_orcl_RAC  --tags patch-vulns config-rac-db.yml

PLAY [dbasm[0]] ****************************************************************

PLAY [dbasm] *******************************************************************

TASK [Gathering Facts] ********************************************************* ok: [g264413522-s314]
ok: [g264413522-s313]

TASK [include_role : patch-vulns] **********************************************

TASK [patch-vulns : Remove known log4j files / folders] ************************
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s313/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s314/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s314/tfa_home/jlib/tfa.war)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s313/tfa_home/jlib/tfa.war)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/lib)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/lib)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/pgx)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/pgx)
ok: [g264413522-s314] => (item=/u01/oracle_install/V982063-01.zip)
ok: [g264413522-s313] => (item=/u01/oracle_install/V982063-01.zip)
ok: [g264413522-s313] => (item=/u01/oracle_install/V982068-01.zip)
ok: [g264413522-s314] => (item=/u01/oracle_install/V982068-01.zip)
changed: [g264413522-s313] => (item=/u01/app/19.3.0/grid/.patch_storage)
changed: [g264413522-s314] => (item=/u01/app/19.3.0/grid/.patch_storage)
changed: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/.patch_storage)
changed: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/.patch_storage)

PLAY RECAP *********************************************************************
g264413522-s313            : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
g264413522-s314            : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

Files used by this session:
	- ./inventory_files/inventory_orcl_RAC
	- ./logs/log_orcl_20230303_195358_RAC.log

```

### CODE REVIEW

File: roles/patch-vulns/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  become_user: root
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/tfa.war"
+    - "{{ grid_home }}/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ grid_home }}/suptools/tfa/release/tfa_home/jlib/tfa.war"
+    - "{{ grid_home }}/.patch_storage"
```
Comment (mfielding) [inline roles/patch-vulns/tasks/main.yml:27]: If all of .patch_storage is removed, users probably won't ever be able to roll back the patch.  Is it possible to just remove the specific files that are known to be an issue?

File: roles/patch-vulns/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+---
+- name: Remove known log4j files / folders
+  become: yes
+  become_user: root
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/tfa.war"
```
Comment (mfielding) [inline roles/patch-vulns/tasks/main.yml:24]: Won't this break TFA even on newer, non-vulnerable TFA installs?

File: roles/patch-vulns/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    - "{{ grid_home }}/suptools/tfa/release/tfa_home/jlib/tfa.war"
+    - "{{ grid_home }}/.patch_storage"
+    - "{{ oracle_home }}/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ oracle_home }}/suptools/tfa/release/tfa_home/jlib/tfa.war"
+    - "{{ oracle_home }}/md/property_graph/lib"
+    - "{{ oracle_home }}/md/property_graph/pgx"
+    - "{{ oracle_home }}/.patch_storage"
+    - "{{ swlib_path }}/V982063-01.zip"
+    - "{{ swlib_path }}/V982068-01.zip"
+  ignore_errors: true
```
Comment (mfielding) [inline roles/patch-vulns/tasks/main.yml:35]: Do we need to ignore errors?  I think state: absent should succeed even if the file doesn't exist.

File: roles/patch-vulns/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    - "{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/tfa.war"
+    - "{{ grid_home }}/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ grid_home }}/suptools/tfa/release/tfa_home/jlib/tfa.war"
+    - "{{ grid_home }}/.patch_storage"
+    - "{{ oracle_home }}/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar"
+    - "{{ oracle_home }}/suptools/tfa/release/tfa_home/jlib/tfa.war"
+    - "{{ oracle_home }}/md/property_graph/lib"
+    - "{{ oracle_home }}/md/property_graph/pgx"
+    - "{{ oracle_home }}/.patch_storage"
+    - "{{ swlib_path }}/V982063-01.zip"
```
Comment (mfielding) [inline roles/patch-vulns/tasks/main.yml:33]: A few potential issues with deleting the base software install:
1. Makes future installs slower as the cached version needs to be re-copied from the GCS bucket
2. Due to 1 it's not particularly effective as the file exists on GCS anyway
3. It's 19c-specific - 18c,12.2 etc base releases are different of course

What do you think about leaving the base release in place, even if it contains vulnerable log4j components?

File: config-rac-db.yml [hunk 5]
```diff
@@ -29,3 +29,10 @@
         loop_var: role_item
       when: create_db|bool
   tags: rac-db-adjustements,rac-db-backups,rac-validation-scripts
+
+- hosts: dbasm
```
Comment (mfielding) [inline config-rac-db.yml:33]: If `--skip-database-config` is specified to set up a cluster without creating a database, is there a way to still ensure we don't have vulnerable components?   (Or, in other words, is there a somewhere we cloud include `patch-vulns` where it is always executed post-install?

**Review by mfielding — COMMENTED** [review]
I like the encapsulation in a new role;  it's a clean way to add functionality.

File: roles/patch-vulns/tasks/main.yml (hunk 2 above)
Comment (mfielding) [reply roles/patch-vulns/tasks/main.yml:24]: Some sample code that extracts the log4j version and can do conditional actions based on the version found:

   - name: List TFA WAR contents
     command: /usr/bin/unzip -l /opt/oracle.ahf/common/jlib/tfa.war
     register: unzipped
   - name: Get version
     debug: msg="{{ unzipped.stdout | regex_findall('log4j-core-[0-9\.]*[0-9]') | first | string }}"
     when: "unzipped.stdout | regex_findall('log4j-core-[0-9\\.]*[0-9]') | first | string is version('log4j-core-2.18.1','<')"

File: roles/patch-vulns/tasks/main.yml (hunk 1 above)
Comment (jcnars) [reply roles/patch-vulns/tasks/main.yml:27]: Done

Adding:
```

$GRID_HOME's patch backup files:
.patch_storage/34419443_Oct_14_2022_05_25_14/files/suptools/tfa.zip
-----------
$ORACLE_HOME's patch backup files:
.patch_storage/34419443_Oct_14_2022_05_25_14/files/md/property_graph.zip
.patch_storage/34419443_Oct_14_2022_05_25_14/files/suptools/tfa.zip

```


Note:
Excerpting from (internal): https://b.corp.google.com/issues/265034396#comment21
> https://paste.googleplex.com/5242743847649280 ==> has the full `superverify` output after the `19.18` run, but the gist is that there is no `.patch_storage/*` file that's vulnerable:
>
```
[root@linuxserver44 ~]#  curl https://www.gstatic.com/log4j_jarscanner/2hamster-scanner-linux -o jar-scanner; chmod +x jar-scanner; ./jar-scanner
...
Found vulnerable JARs:
/u01/oracle_install/V982063-01.zip
/u01/oracle_install/V982068-01.zip
```

File: roles/patch-vulns/tasks/main.yml (hunk 2 above)
Comment (jcnars) [reply roles/patch-vulns/tasks/main.yml:24]: As observed in:
*  in the section (internal) titled `Vulnerability introduction based on install option`: https://docs.google.com/document/d/1gPYY4mvovxdqSPjqAPSZbSVHbDfilpTmdIvZSIhR6Kw/edit#heading=h.2uf217nh4ra8
* and in bug update (internal): https://b.corp.google.com/issues/265034396#comment21

the `{{ grid_home }}/tfa/{{ inventory_hostname_short }}/tfa_home/jlib/tfa.war` only gets introduced in `--no-patch` installations. In this case, in order to not introduce code complexity & maintainability, if older jars are present in the TFA, removing vulnerable jars may be considered to hold higher priority than usability of TFA.

Based on this reasoning, I would suggest, we retain the existing logic.

File: roles/patch-vulns/tasks/main.yml (hunk 3 above)
Comment (jcnars) [reply roles/patch-vulns/tasks/main.yml:35]: Done, removed the line

File: roles/patch-vulns/tasks/main.yml (hunk 4 above)
Comment (jcnars) [reply roles/patch-vulns/tasks/main.yml:33]: Given the objective of this KR itself (rid log4j vulnerable jars/wars from BMS host), my thoughts are along the lines of:
* weighing the inconvenience of GCS /NFS copying again for a occasional re-invocation of `install-oracle.sh` for a failed run against
* ensuring the BMS host is wiped clean of any log4j  vulnerabilities
* I tend to lean towards removing the zip files.

What are your thoughts on this line of reasoning?

(Cleaning files existing on other non-BMS-host locations (like GCS or other NFS repositories) is a non-goal of this KR.)

**Comment by jcnars** [comment]
I have tested the refactored code that incorporates the comments as follows (now sourced from `install-sw.yml`):
```
✔ ~/mydrive/bmaas/log4j_remediation/bms-toolkit [log4j_kr_remediation|✚ 15…5] 
07:53 $  ./install-oracle.sh --ora-swlib-bucket gs://bmaas-testing-oracle-software --instance-ssh-user ansible1 --instance-ssh-key ~/.ssh/id_rsa_bms_toolkit.ansible1_in_poc --instance-ssh-extra-args ''\''-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentityAgent=no -o IdentitiesOnly=true -o ProxyCommand="ssh -W %h:%p -q jcnarasimhan@control-node.mfild.com"'\'''  --backup-dest /u01/backups --ora-swlib-path /u01/oracle_install --ora-version 19 --ora-swlib-type gcs --ora-asm-disks workshop_asm_disk_config.json --ora-data-mounts workshop_data_mounts_config.json --cluster-type RAC --cluster-config workshop_cluster_config.json --ora-data-diskgroup DATA --ora-reco-diskgroup RECO --ora-db-name orcl --ora-db-container false  --no-patch  -- '--tags patch-vulns'      >> /usr/local/google/home/jcnarasimhan/mydrive/bmaas/workshop/bms-toolkit/logs/log4remediation_3

✔ ~/DriveFS/My Drive/bmaas/log4j_remediation/bms-toolkit/logs [log4j_kr_remediation|✚ 19…5] 
10:20 $ tail -f /usr/local/google/home/jcnarasimhan/mydrive/bmaas/workshop/bms-toolkit/logs/log4remediation_3
PLAY [dbasm] *******************************************************************

PLAY RECAP *********************************************************************


Running Ansible playbook: ansible-playbook -i ./inventory_files/inventory_orcl_RAC  --tags patch-vulns prep-host.yml

PLAY [Populate variables] ******************************************************

PLAY [dbasm] *******************************************************************

TASK [Gathering Facts] *********************************************************
ok: [g264413522-s313]
ok: [g264413522-s314]

PLAY [dbasm[0]] ****************************************************************

TASK [Gathering Facts] *********************************************************
ok: [g264413522-s313]

PLAY RECAP *********************************************************************
g264413522-s313            : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
g264413522-s314            : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   


Running Ansible playbook: ansible-playbook -i ./inventory_files/inventory_orcl_RAC  --tags patch-vulns install-sw.yml

PLAY [Populate variables] ******************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm[0]] ****************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm] *******************************************************************

PLAY [dbasm[0]] ****************************************************************

PLAY [dbasm] *******************************************************************

TASK [Gathering Facts] *********************************************************
ok: [g264413522-s314]
ok: [g264413522-s313]

TASK [include_role : patch-vulns] **********************************************

TASK [patch-vulns : Remove known log4j files / folders] ************************
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s314/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s313/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s313/tfa_home/jlib/tfa.war)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/tfa/g264413522-s314/tfa_home/jlib/tfa.war)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s314] => (item=/u01/app/19.3.0/grid/.patch_storage/34419443_Oct_14_2022_05_25_14/files/suptools/tfa.zip)
ok: [g264413522-s313] => (item=/u01/app/19.3.0/grid/.patch_storage/34419443_Oct_14_2022_05_25_14/files/suptools/tfa.zip)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/log4j-core-2.9.1.jar)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/suptools/tfa/release/tfa_home/jlib/tfa.war)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/lib)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/lib)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/pgx)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/md/property_graph/pgx)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/.patch_storage/34419443_Oct_14_2022_05_25_14/files/md/property_graph.zip)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/.patch_storage/34419443_Oct_14_2022_05_25_14/files/md/property_graph.zip)
ok: [g264413522-s313] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/.patch_storage/34419443_Oct_14_2022_05_25_14/files/suptools/tfa.zip)
ok: [g264413522-s314] => (item=/u01/app/oracle/product/19.3.0/dbhome_1/.patch_storage/34419443_Oct_14_2022_05_25_14/files/suptools/tfa.zip)
ok: [g264413522-s313] => (item=/u01/oracle_install/V982063-01.zip)
ok: [g264413522-s314] => (item=/u01/oracle_install/V982063-01.zip)
ok: [g264413522-s313] => (item=/u01/oracle_install/V982068-01.zip)
ok: [g264413522-s314] => (item=/u01/oracle_install/V982068-01.zip)

PLAY RECAP *********************************************************************
g264413522-s313            : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
g264413522-s314            : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   


Running Ansible playbook: ansible-playbook -i ./inventory_files/inventory_orcl_RAC  --tags patch-vulns config-rac-db.yml

PLAY [dbasm[0]] ****************************************************************

PLAY RECAP *********************************************************************


Files used by this session:
	- ./inventory_files/inventory_orcl_RAC
	- ./logs/log_orcl_20230329_102016_RAC.log


```

File: roles/patch-vulns/tasks/main.yml (hunk 1 above)
Comment (mfielding) [reply roles/patch-vulns/tasks/main.yml]: Looks good, thanks

File: roles/patch-vulns/tasks/main.yml (hunk 2 above)
Comment (mfielding) [reply roles/patch-vulns/tasks/main.yml:24]: Ahh, if `tfa.war` no longer exists in recent PSUs, that addresses my original question.

It does bring up another related question though:  as far as I recall, we've tested the 19.3 base release and 19.18 PSU.  What about the releases in between?  Do we believe removing `log4j-core-2.9.1` and `tfa.war` to be sufficient?

File: config-rac-db.yml (hunk 5 above)
Comment (mfielding) [reply config-rac-db.yml]: LGTM;  `install-sw.yml` definitely sounds like the right place to do this.

**Review by mfielding — APPROVED** [review]

File: roles/patch-vulns/tasks/main.yml (hunk 2 above)
Comment (jcnars) [reply roles/patch-vulns/tasks/main.yml:24]: I will need to do actual installs to verify that, but I remember these files go away with the OCT-2021 PSU.

**Pull request merged**